src/main/java/org/skriptlang/skript/bukkit/lang/eventvalue/EventValueRegistry.java

```diff
@@ -0,0 +1,253 @@
+package org.skriptlang.skript.bukkit.lang.eventvalue;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.SkriptAPIException;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Range;
+import org.jetbrains.annotations.Unmodifiable;
+import org.skriptlang.skript.lang.converter.Converters;
+import org.skriptlang.skript.util.Registry;
+import org.skriptlang.skript.util.ViewProvider;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Registry and resolver for {@link EventValue} definitions.
+ * <p>
+ * Use this registry to register, unregister and resolve event values by identifier text or by
+ * desired value type. Resolution prefers the closest matching event type and, optionally,
+ * can fall back to default time state and/or allow value-type conversion.
+ * <p>
+ * Obtain an instance using {@code SkriptAddon#registry(EventValueRegistry.class)}.
+ * <br>
+ * Or an unmodifiable view using {@code Skript.instance().registry(EventValueRegistry.class)}.
+ */
+public interface EventValueRegistry extends Registry<EventValue<?, ?>>, ViewProvider<EventValueRegistry> {
+
+	/**
+	 * Creates an empty event value registry.
+	 */
+	static EventValueRegistry empty(Skript skript) {
+		return new EventValueRegistryImpl(skript);
+	}
+
+	/**
+	 * When resolving a value for {@link EventValue#TIME_PAST}
+	 */
+	int FALLBACK_TO_DEFAULT_TIME_STATE = 0b01;
+	/**
+	 * Allow converting the resolved value type using {@link Converters}
+	 */
+	int ALLOW_CONVERSION = 0b10;
+	/**
+	 * Default combination of resolution flags: fallback to default time and allow conversion.
+	 */
+	int DEFAULT_RESOLVE_FLAGS = FALLBACK_TO_DEFAULT_TIME_STATE | ALLOW_CONVERSION;
```

Comment:
it would be nice but not critical to have these as an enum with a more standard api
Eg addFlag, hasFlag, setFlags(...) 
I think an enumset is often implemented as a bitmap behind the scenes anyway